### PR TITLE
Feature: Add specs for ApiVersion

### PR DESCRIPTION
**Pull request by bohme**
Hunting those 100% 🤠

### CODE REVIEW

File: lib/constraints/api_version.rb [hunk 1]
```diff
@@ -9,8 +9,9 @@ def initialize(version)
     end
 
     def matches?(request)
-      request.headers["X-Api-Version"].present? &&
-        request.headers["X-Api-Version"] == version
+      return false unless request&.headers
+
+      request.headers["X-Api-Version"].presence == version
```
Comment (GeoffAbtion) [inline lib/constraints/api_version.rb:14]: Do you need `.presence` here?

File: lib/constraints/api_version.rb [hunk 2]
```diff
@@ -9,8 +9,9 @@ def initialize(version)
     end
 
     def matches?(request)
-      request.headers["X-Api-Version"].present? &&
-        request.headers["X-Api-Version"] == version
+      return false unless request&.headers
```
Comment (GeoffAbtion) [inline lib/constraints/api_version.rb:12]: &. 😿

File: spec/lib/constaints/api_version_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    context "when no headers present" do
+      it "returns false" do
+        version = "1.33.7"
+        version_constraints = Constraints::ApiVersion.new(version)
+
+        expect(version_constraints.matches?(nil)).to eq(false)
+      end
+    end
+  end
+end
```
Comment (GeoffAbtion) [inline spec/lib/constaints/api_version_spec.rb:36]: Needs a test for `X-Api-Version present but does not match version attribute`.

File: lib/constraints/api_version.rb (hunk 1 above)
Comment (bohme) [reply lib/constraints/api_version.rb]: guess I don't 🍡

File: spec/lib/constaints/api_version_spec.rb [hunk 4]
```diff
@@ -14,6 +14,16 @@
       end
     end
 
+    context "when 'X-Api-Version' header doesn't matche version attribute" do
+      it "returns true " do
```
Comment (GeoffAbtion) [inline spec/lib/constaints/api_version_spec.rb:18]: `matche` => `match`
`true` => `false`

**Comment by bohme** [comment]
ohh you can taste the sweet sweet 100% from here:
```
Randomized with seed 32343
................................
Finished in 8.05 seconds (files took 7.06 seconds to load)
32 examples, 0 failures
Randomized with seed 32343
Coverage report generated for RSpec to /app/coverage. 81 / 81 LOC (100.0%) covered.
```

**Pull request merged**